packages/otelbin/src/components/react-flow/layout/getSimpleCycles.ts

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// The types of dagre API is incorret so we have to disable type check for this file.
+//
+// eslint-disable-next-line
+// @ts-nocheck
+
+import Dagre from "@dagrejs/dagre";
+
+// getSimpleCycles use Johnson algorithm to find all the simple cycles of the graph.
+// ref: https://www.cs.tufts.edu/comp/150GA/homeworks/hw1/Johnson%2075.PDF
+export function getSimpleCycles(g: Dagre.graphlib.Graph) {
```

Comment:
**Question:** For licensing reasons, is this a code a copy from some location? It somehow looks like it, especially with the different casing of `johnson_simple_cycles`?

Reply:
It is not copied from somewhere else. I found some python and C++ reference code and this code is basically written based on the pseudo code in reference pdf.